SU2_CFD/src/variables/CNEMOEulerVariable.cpp

```diff
@@ -143,23 +144,20 @@ CNEMOEulerVariable::CNEMOEulerVariable(su2double val_pressure,
   UnderRelaxation.resize(nPoint) = su2double(1.0);
   LocalCFL.resize(nPoint) = su2double(0.0);
 
+  /*--- Set mixture state ---*/
+  fluidmodel->SetTDStatePTTv(val_pressure, val_massfrac, val_temperature, val_temperature_ve);
+
+  /*--- Compute necessary quantities ---*/
+  rho = fluidmodel->GetDensity();
+  soundspeed = fluidmodel->ComputeSoundSpeed();
+  for (iDim = 0; iDim < nDim; iDim++){
+    sqvel += val_mach[iDim]*soundspeed * val_mach[iDim]*soundspeed;
+  }
+  energies = fluidmodel->ComputeMixtureEnergies();
```

Comment:
```suggestion
  const su2double rho = fluidmodel->GetDensity();
  const su2double soundspeed = fluidmodel->ComputeSoundSpeed();
  const su2double sqvel = GeometryToolbox::SquaredNorm(nDim, val_mach) * pow(soundspeed,2);
  const auto& energies = fluidmodel->ComputeMixtureEnergies();
```
You may need the include for geometry_toolbox.hpp.
Be aware of the methods in CNEMOGas that return by reference, those little copies of std::vector add up and it is an easy fix.

Reply:
That is true, and may be the main reason of the slowness of SU2-NEMO iterations. I will try to address that in my next PR.

Reply:
:+1: You can have a look at #716 (scroll down) for how to use perf-tools to profile the code and point you to the hot spots "first rule of performance is that you need to measure it". There should be hotter areas in NEMO than vector copies.